### PR TITLE
In-app for text recognition

### LINKED ISSUE #247: In-app purchase
We decided to change economic and add in-apps for tools inside the app. Text recognition will be the first in-app.

## Feature
- [x] UI
- [x] Register in-app
- [x] Fetch price
- [x] Purchase
- [x] Restore
- [x] UI management
- [x] Handle delay during operations
- [x] Present errors
- [x] Unlock and persist
   - [x] Move to keychain, `.plist` can be edited manually. [Sample](https://acerodstin.ru/ru-ru/articles/разработка-propertywrapper-для-keychain)
- [x] Link unlock status with checkbox (move recognition offer to label's controller to keep in single place). Just removed checkbox and create #249 
- [x] Upgrade users that already bought full version
- [x] Add to container's UI
- [x] Shorten label of activation button

## Details
- [x] Launch recognition right after purchase
- [x] Load prices on start without waiting for UI 
   - [x] Upgrade to in-app on document's open, without pressing the restore button
   - [ ] Do it once per launch, not once per document
   - [ ] Log to analytics when upgrade had happened?
- [ ] Should I check that environment is authorised for payment?
- [ ] Coloured border of purchase should react to window deactivation

## Infra
- [x] Join Preview with Designer to share purchases
- [ ] [Test deferred transactions](https://stackoverflow.com/questions/25510678/how-to-test-skpaymenttransactionstatedeferred)
- [ ] Unit-tests? Fine to have a test that fetch product before payment
- [ ] Flag that allows to test in-app in Debug

### PR DESCRIPTION
**Pull request by akaDuality**
Closes #247 

Screenshot 2023-03-25 at 22 19 04

### CODE REVIEW

File: Shared/Sources/Purchases/PurchaseRepository.swift [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        for await verification in Transaction.currentEntitlements {
+            if case let .verified(transaction) = verification {
+                print("Found transaction to restore \(transaction.productID)")
+                await unlockAndFinish(transaction)
+            }
+        }
+    }
+    
+    @available(macOS 13.0, *)
+    func migrateAppPurchaseToFullUnlock() {
```
Comment (drucelweisse) [inline Shared/Sources/Purchases/PurchaseRepository.swift:105]: Why is that function limited to macOS 13?

File: Shared/Package.swift [hunk 2]
```diff
@@ -109,6 +111,18 @@ let package = Package(
         .executableTarget(name: "SamplesStructure", dependencies: [
             "Samples",
             .product(name: "ArgumentParser", package: "swift-argument-parser"),
-        ])
+        ]),
+        
+        .target(
+            name: "Purchases",
+            dependencies: [
+                "Document",
```
Comment (drucelweisse) [inline Shared/Package.swift:119]: Is Purchases package is dependent on Document?

**Review by drucelweisse — CHANGES_REQUESTED** [review]
Cool! Liked, that it is reusable for any kind of product. Also have some questions

File: Shared/Sources/Purchases/PurchaseRepository.swift (hunk 1 above)
Comment (akaDuality) [reply Shared/Sources/Purchases/PurchaseRepository.swift:105]: `currentEntitlements` only on iOS 13

File: Shared/Package.swift (hunk 2 above)
Comment (akaDuality) [reply Shared/Package.swift:119]: O, thank you: I got UserDefaults wrapper from Document, but migrate to iCloud instead and forgot to remove this

**Pull request merged**